app/src/main/java/com/infomaniak/mail/utils/Extensions.kt

```diff
@@ -366,19 +368,30 @@ fun Fragment.createInputDialog(
     @StringRes title: Int,
     @StringRes hint: Int,
     @StringRes confirmButtonText: Int,
-    onErrorCheck: (CharSequence) -> String?,
+    onErrorCheck: suspend (CharSequence) -> String?,
     onPositiveButtonClicked: (CharSequence?) -> Unit,
 ) = with(DialogInputBinding.inflate(layoutInflater)) {
 
+    var errorJob: Job? = null
+
+    fun Button.checkValidation(text: CharSequence) {
+        errorJob?.cancel()
+        errorJob = lifecycleScope.launch(Dispatchers.IO) {
+            val error = onErrorCheck(text.trim())
+            if (errorJob?.isActive == true) Dispatchers.Main {
+                textInputLayout.error = error
+                isEnabled = error == null
+            }
+        }
+    }
+
     fun AlertDialog.setupOnShowListener() = apply {
         setOnShowListener {
             showKeyboard()
             getButton(AlertDialog.BUTTON_POSITIVE).apply {
                 isEnabled = false
                 textInput.doAfterTextChanged {
-                    val error = if (it.isNullOrBlank()) null else onErrorCheck(it.trim())
-                    isEnabled = it?.isNotBlank() == true && error == null
-                    textInputLayout.error = error
+                    if (it.isNullOrBlank()) isEnabled = false else checkValidation(it.trim())
```

Comment:
The layout error is not hidden anymore if the field is empty in one change, nor when the popup is dismissed

Reply:
Done ☑